Control Plane latency recommendations for reliable clusters (#76677)
openshift/openshift-docs: modules/control-plane-latency.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/control-plane-latency-recommendations-for-reliable-clusters.adoc

:_mod-docs-content-type: CONCEPT
[id="control-plane-latency_{context}"]
= Recommended Control Plane latency for reliable clusters

> It's recommended to ensure that the latency between each of the control plane nodes is within 15ms to ensure performant and reliable cluster. Some of the metrics to keep track of includes etcd gRPC requests latency, fysnc latency and any critical alerts. Here are the PromQL queries:

Reviewer: ```suggestion
Latency between each of the control plane nodes must be less than 15ms to ensure a well performing and reliable cluster. Some of the metrics to keep track of include etcd gRPC requests latency, fysnc latency and any critical alerts. Here are the PromQL queries:
```

Reviewer: Great work @shahsahil264 ! I have a few small items:
- squash your commits and push to origin.
- Tell us what versions of OpenShift this applies to? Typically 4.12+ Add that to your comment 0, the first comment in the PR so we can see it easily.
- When you re-push as I have suggested above, PROW will generate a new preview link. Please put that in Comment 0 if you can please sir. I could not find your updates in the preview that PROW had already generated. 
- If you are working from a JIRA, please add that link to Comment 0
I am excited, @shahsahil264 , your work is a wonderful addition to the docs! 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

// Module included in the following assemblies:
//
// * scalability_and_performance/control-plane-latency-recommendations-for-reliable-clusters.adoc

:_mod-docs-content-type: CONCEPT
[id="control-plane-latency_{context}"]
= Recommended Control Plane latency for reliable clusters

> We recommend inducing 15ms latency, for the cluster to function normally and to avoid critical alerts such as KubeAPIErrorBudgetBurn and etcdGRPCRequestsSlow.

Reviewer 2: Maybe reword it to: 
```
It's recommended to ensure that the latency between each of the control plane nodes is within 15ms to ensure performant and reliable cluster. Some of the metrics to keep track of includes etcd grpc requests latency, fysnc latency and any critical alerts. Here are the PromQL queries:
histogram_quantile(0.99, sum(rate(grpc_server_handling_seconds_bucket{job=~".*etcd.*", grpc_method!="Defragment", grpc_type="unary"}[5m])) without(grpc_type)) < 0.15
avg_over_time(histogram_quantile(0.99, rate(etcd_disk_wal_fsync_duration_seconds_bucket[2m]))[10m:]) < 1
ALERTS{severity="critical", alertstate="firing"} > 0
```
Thoughts?

Author: Sounds good! I'll make the changes
